platforms/android/library/src/main/java/io/element/android/wysiwyg/spans/CodeBlockSpan.kt

```diff
@@ -0,0 +1,89 @@
+package io.element.android.wysiwyg.spans
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Paint
+import android.graphics.Rect
+import android.os.Parcel
+import android.text.Layout
+import android.text.ParcelableSpan
+import android.text.TextPaint
+import android.text.style.BackgroundColorSpan
+import android.text.style.LeadingMarginSpan
+import android.text.style.MetricAffectingSpan
+import android.text.style.ParagraphStyle
+import android.text.style.TypefaceSpan
+import android.text.style.UpdateAppearance
+import androidx.core.content.ContextCompat
+
+/**
+ * Code block (```some code``` in Markdown, <pre> in HTML) Span that applies a monospaced font style
+ * and adds a background color to a whole paragraph.
+ */
+class CodeBlockSpan : MetricAffectingSpan, LeadingMarginSpan, UpdateAppearance {
```

Comment:
I believe the idea was to use the same spans (or at least have the same visual representation) in both the editor and the timeline messages, that's why these are public. 

However, maybe it might be worth it to create a different lib/module just for shared spans for the editor and clients? Or maybe we can just copy & paste the implementations if needed.

Reply:
Ok, in that case a shared lib/module makes sense (or just exposing it through this library as we are doing it here)

Reply:
I actually made them internal in 5563492. Should I revert that?

Reply:
Yeah, I think you had a good point about making these formatting spans available to apps that use the library (whatever form that takes)